pkg/cmd/playback.go

```diff
@@ -0,0 +1,245 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+
+	"github.com/stripe/stripe-cli/pkg/playback"
+	"github.com/stripe/stripe-cli/pkg/validators"
+)
+
+const defaultPort = 13111
+const defaultWebhookPort = 13112
+const endpointsDocString = `
+--- Controlling the server ---
+You can configure the running server instance via HTTP GET endpoints (prefixed with "/pb/").
+
+--- List of server control endpoints ---
+GET /pb/mode/[mode]
+Sets the server mode to one of ["auto", "record", "replay"]
+
+GET /pb/cassette/setroot?dir=[path_to_directory]
+Set the root directory for reading/writing cassettes. All cassette paths are relative to this directory.
+
+GET /pb/cassette/load?filepath=[filepath]
+Load the cassette file at the given filepath, relative to the cassette root directory.
+
+GET /pb/cassette/eject
+Eject (unload) the current cassette and do any teardown. In record mode, this writes the recorded interactions to the cassette file.
+`
+
+type playbackCmd struct {
+	cmd *cobra.Command
+
+	mode string
+
+	apiBaseURL     string
+	filepath       string
+	cassetteDir    string
+	address        string
+	webhookAddress string
+	serveHTTPS     bool
+}
+
+func newPlaybackCmd() *playbackCmd {
+	pc := &playbackCmd{}
+
+	pc.cmd = &cobra.Command{
+		// This flag doesn't actually hide the command because we use a template to display the --help text
+		// To unhide this command, see getUsageTemplate() in templates.go, where we explicitly hide the `playback` cmd.
+		Hidden: true,
+		Use:    "playback",
+		Args:   validators.NoArgs,
+		Short:  "Start a `playback` server",
+		Long: `
+--- Overview ---
+The playback command starts a local proxy server that intercepts outgoing requests to the Stripe API.
+
+There are three modes of operation:
+
+"record": Any requests received are forwarded to the api.stripe.com, and the response is returned. All interactions
+are written to the loaded 'cassette' file for later playback in replay mode.
+
+"replay": All received requests are terminated at the playback server, and responses are played back[1] from a cassette file. A existing cassette most be loaded.
+
+"auto": The server determines whether to run in "record" or "replay" mode on a per-cassette basis. If the cassette exists, operates in "replay" mode. If not, operates in "record" mode.
```

Comment:
```suggestion
"auto" (default): The server determines whether to run in "record" or "replay" mode on a per-cassette basis. If the cassette exists, operates in "replay" mode. If not, operates in "record" mode.
```

Reply:
Oh, I guess can we make auto the default?

Reply:
Yep!